lms/djangoapps/mobile_api/users/serializers.py

```diff
@@ -34,6 +35,10 @@ def to_representation(self, course_overview):
                 kwargs={'course_id': course_id},
                 request=request
             )
+            discussion_url = reverse(
+                'discussion_course',
+                kwargs={'course_id': course_id},
+                request=request) if course_overview.is_discussion_tab_enabled() else None
```

Comment:
You need to use the `UrlResetMixin` class to make sure the `urls.py` module gets re-evaluated after the Discussion feature flag is enabled:

```
from util.testing import UrlResetMixin
...
class TestUserEnrollmentApi(UrlResetMixin, MobileAPITestCase, MobileAuthUserTestMixin):
```